kie-server-parent/kie-server-maven-plugin/src/main/java/org/kie/server/gateway/KieServerGateway.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2017 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+package org.kie.server.gateway;
+
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.project.MavenProject;
+import org.kie.internal.runtime.conf.RuntimeStrategy;
+import org.kie.server.api.marshalling.MarshallingFormat;
+import org.kie.server.api.model.*;
+import org.kie.server.client.KieServicesClient;
+import org.kie.server.client.KieServicesConfiguration;
+import org.kie.server.client.KieServicesFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class KieServerGateway {
+
+    public static final Logger LOG = LoggerFactory.getLogger(KieServerGateway.class);
+
+    private final KieServicesConfiguration config;
+    private KieServicesClient client;
+
+    public KieServerGateway(String protocol, String hostname, Integer port, String username, String password, Integer timeout, String contextPath) {
+
+        String serverUrl = new StringBuilder(protocol)
+                .append("://")
+                .append(hostname)
+                .append(":")
+                .append(port)
+                .append("/")
+                .append(contextPath)
+                .append("/services/rest/server")
+                .toString();
+
+        LOG.info("Server Url {}", serverUrl);
+
+        config = KieServicesFactory.newRestConfiguration(serverUrl, username, password);
+        config.setMarshallingFormat(MarshallingFormat.JSON);
+        config.setTimeout(timeout);
+    }
+
+    public void init() throws MojoFailureException {
+
+        try {
+            client = KieServicesFactory.newKieServicesClient(config);
+        } catch (RuntimeException kieEx) {
+            throw new MojoFailureException("error on enstablish connection with remote server: " + kieEx.getMessage());
+        }
+
+    }
+
+    public void deploy(MavenProject project, String container, RuntimeStrategy strategy) throws MojoFailureException {
+
+        KieContainerResource kieContainer = new KieContainerResource();
+        kieContainer.setContainerId(container);
+
+        ReleaseId releaseId = new ReleaseId();
+        releaseId.setGroupId(project.getGroupId());
+        releaseId.setArtifactId(project.getArtifactId());
+        releaseId.setVersion(project.getVersion());
+        kieContainer.setReleaseId(releaseId);
+
+        KieServerConfigItem configItem = new KieServerConfigItem();
```

Comment:
I am thinking that we should make this optional and add this config property when it is given otherwise it can easily replace one set in jar's deployment descriptor. wdyt?